rustls/src/client/builder.rs

```diff
@@ -163,6 +163,34 @@ impl ConfigBuilder<ClientConfig, WantsClientCert> {
         Ok(self.with_client_cert_resolver(Arc::new(resolver)))
     }
 
+    /// Sets a raw public key and matching private key for use
+    /// in client authentication.
+    ///
+    /// `key_der` is a DER-encoded private key as PKCS#1, PKCS#8, or SEC1. The
+    /// `aws-lc-rs` and `ring` [`CryptoProvider`]s support all three encodings,
+    /// but other `CryptoProviders` may not.
+    ///
+    /// This function fails if `key_der` is invalid, or if the public key cannot be retrieved from the private key.
+    ///
+    /// Note: This API is only usable for `SigningKey` implementations that can produce the corresponding public key.
+    /// Please ensure that the `CryptoProvider` you are using supports this functionality.
+    pub fn with_client_auth_raw_key(
```

Comment:
Instead of adding this niche API surface, should we just expose `AlwaysResolvesClientRawPublicKeys` directly (maybe under a better name, even if it isn't consistent with the existing naming)?

Reply:
@ctz got opinions on this?

Reply:
I agree about exposing the `handy::AlwaysResolvesClientRawPublicKey` exemplar implementation, mainly because I have duplicates of `AlwaysResolvesClientCert` and `AlwaysResolvesChain` in downstream projects for using alternate private key types (and have been too lazy thus far to fix that...).

I have a (very marginal) preference for losing the builder function after that, on the basis of a insubstantial worry that it could funnel new users towards this feature rather than away from it?

Reply:
> I have a (very marginal) preference for losing the builder function after that, on the basis of a insubstantial worry that it could funnel new users towards this feature rather than away from it?

That was one of my reasons as well.

Reply:
Gotcha, `handy::AlwaysResolvesClientRawPublicKey` and `handy::AlwaysResolvesServerRawPublicKey` are now public and the builder functions are removed.